kernel/memory/src/paging/mapper.rs

```diff
@@ -805,6 +805,10 @@ impl MappedPages {
             );
         }
 
+        if size_in_bytes >= isize::MAX as usize {
+            return Err("MappedPages::as_slice(): length * size_of::<T>() must be less than isize::MAX");
```

Comment:
```suggestion
        if size_in_bytes > isize::MAX as usize {
            return Err("MappedPages::as_slice(): length * size_of::<T>() must be no larger than isize::MAX");
```

Reply:
wow, must be tired... i misread the req from the `slice::from_raw_parts` docs